### PR TITLE
GDAX cancelOrder fix and misc other fixes

### PR DESCRIPTION
**Pull request by werkkrew**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Big Fix / Feature

* **What is the current behavior?** (You can also link to an open issue here)
Same as PR #1806 cancelOrder does not detect filled orders. 

* **What is the new behavior (if this is a feature change)?**
Calls to cancelOrder that result in order not found error will presume the order was filled and return true to portfolioManager

Additional minor feature add is to have the ability to enable/disable sandbox using a config value instead of a hard coded boolean in the exchange file itself.

* **Other information**:
Also same as #1806 I am not 100% this approach is 100% as robust as it could/should be but it works as a stop-gap.

@askmike @cmroche

### CODE REVIEW

**Comment by askmike** [comment]
ACK all your changes, would you mind reverting all the whitespace changes though? (2 space indents and the likes)? Thanks!

**Comment by werkkrew** [comment]
Cleaned up the whitespace a good bit to clarify the changes introduced.

**Comment by askmike** [comment]
:+1: thanks!

File: exchanges/gdax.js [hunk 1]
```diff
@@ -204,10 +213,15 @@ Trader.prototype.getOrder = function(order, callback) {
...
 Trader.prototype.cancelOrder = function(order, callback) {
-  var result = function(err, data) {
-    // todo, verify result..
-    callback();
-  };
+    // callback for cancelOrder should be true if the order was already filled, otherwise false
+    var result = function(err, data) {
+        if(err) {
+            log.error('Error cancelling order:', err);
+            callback(true);  // need to catch the specific error but usually an error on cancel means it was filled
```
Comment (cmroche) [inline exchanges/gdax.js:220]: If you don't return here you will also call callback(false).

**Review by cmroche — COMMENTED** [review]
Small bug

File: exchanges/gdax.js (hunk 1 above)
Comment (werkkrew) [reply exchanges/gdax.js:220]: Since it was already merged can you implement this fix?

Comment (werkkrew) [reply exchanges/gdax.js:220]: I updated my branch with the returns...

**Pull request merged**